### PR TITLE
Feat/pod level opa

### PR DESCRIPTION
**Pull request by vlad-tokarev**
* Fresh docs according changes in security components
* Delete keycloak configuration info as not related to ODAHU system

Updated images:
image

image

New images:
image

### CODE REVIEW

File: source/comp_feedback.rst [hunk 1]
```diff
@@ -0,0 +1,46 @@
...
+and pre-defined OPA policies. If :ref:`Security Subsystem <comp_security:Security-subsystem>` is installed then all requests
+to :term:`Model Feedback API` service will be enforced using :ref:`pre-defined OPA policies <gen_security:ODAHU API and Feedback aggregator policies>`.
+
+Implementation details
+----------------------
+
+:term:`Feedback aggregator` contains two major subcomponents
+
+- REST Server that provides :term:`Model Feedback API` and send them to configured fluentd server
+- Envoy Proxy tap filter that catch all requests and response to deployed models and send this info
```
Comment (karbyshevds) [inline source/comp_feedback.rst]:
```suggestion
- Envoy Proxy tap filter that catch all requests and response of deployed models and send this info
```

File: source/comp_security.rst [hunk 2]
```diff
@@ -0,0 +1,50 @@
+
+======================
+Security subsystem
+======================
+
+Security subsystem is distributed as a helm chart and relies on `OpenPolicyAgent`_ to make decisions about authorization
+and `Istio`_ to enforce authorization for requests to the protected services.
+
+You can read concepts about ODAHU security in :ref:`an appropriate docs section<gen_security:Security>`
```
Comment (karbyshevds) [inline source/comp_security.rst]:
```suggestion
You can read about ODAHU security concepts in :ref:`an appropriate docs section<gen_security:Security>`
```

File: source/comp_security.rst [hunk 3]
```diff
@@ -0,0 +1,50 @@
...
+  to force authentication and authorization for all incoming http requests
+- ConfigMap with common policy that included into all `OpenPolicyAgent`_ sidecars and implements masking of a sensitive data
+- ConfigMap with default policy that included into `OpenPolicyAgent`_ sidecars when pod does not specify ConfigMap with its polices
+
+When webhook server deploys `OpenPolicyAgent`_ sidecar it attaches policies from ConfigMap. ConfigMap can be found by
+value of pod label ``opa-policy-config-map-name``. If this label is missed then default policy will be used.
+Default policy – reject all requests.
+
+Different ODAHU components such as :ref:`API <comp_api:API>` and :ref:`Feedback aggregator <comp_feedback:Feedback aggregator>`
+are distributed with a pre-defined set of OpenPolicyAgent_ policies. They create ConfigMap with their policies in the deployment process.
```
Comment (karbyshevds) [inline source/comp_security.rst]:
```suggestion
are distributed with a pre-defined set of OpenPolicyAgent_ policies. They create ConfigMap with their policies during the deployment process.
```

File: source/gen_security.rst [hunk 4]
```diff
@@ -344,22 +280,32 @@ use more general `Access based access control`_. For this rewrite `core.rego` fi
...
 - lines 8-16: allow access if there are required permissions for action and resource for at least one user's roles
-- lines 12-21: allow access to root for any user
+- lines 18-21: allow access to root for any user
 - lines 23-26: allow access to swagger docs to any user
 
+
+ODAHU ML Models pods policies
+""""""""""""""""""""""""""""""""""""""
+
+All deployed models automatically have policies that permit requests to them to all users
```
Comment (karbyshevds) [inline source/gen_security.rst]:
```suggestion
All deployed models contain default policies that permit requests to them to all users
```

File: source/gen_security.rst [hunk 5]
```diff
@@ -344,22 +280,32 @@ use more general `Access based access control`_. For this rewrite `core.rego` fi
...
+that have :term:`Model Deployment Access Role Name`. This role can be set while deployment of a model using
```
Comment (karbyshevds) [inline source/gen_security.rst]:
```suggestion
that have :term:`Model Deployment Access Role Name`. This role can be set on model deployment stage using
```

File: source/img/security_comp_impl.drawio [hunk 6]
```diff
@@ -0,0 +1 @@
...eeiUDhpd5glHo5GwJsRf68wgk8xhEohtIEtHe0F0pkEqIH4FYP0I6MBoBMIdQoPKJhgwY3egBjK5OhZhq4AZkVknCCBumOKZz7gLH2GUcJjU21ihiXHa2Yk0/td/RTxI0MQmS/N9m5Scf7Moyq/p2L0Yva3mzdgdJyFbPlbJVPhSCqVUYXkRALzFqk8RqK0JA3mBIeAyWHg/lVEXvbFwi6L15ZwdWKOcwbsRnXFYkPtwMlLHN4GMusHJAtQv8lrKNm4uB4H1+/g7w9IFXtWldjK1qQ21VPS9b1SRb/ac4a9yXZ42RbXXsQ6Hd9aMnN1zVOunOrVyZrb37kzdufSgKjPNCgS6hYOVQLFt/GoCEF7MIlQx1rH/Lj3d3OA3LI9p2hynF0dHDQEN5OKOIBcabQ35RmcjNa904t+e08KlYsS/MyRtDzXtxXuZtSOYtCb4hoIOTPnj4tO3hOxkkzV6Y11pfBskr/liLiKAZ0Vyzj3KlKCY74bOjhc6/ZuaWn/d5g1qQtQ512UsthuhyLo8MvCh6s/aeQdQOTbM7vOzDlifAH+19fhFx9ZDtIImZ40yvQodjfp2Qb0VhC5JwKgx3cmTqQgax2QPiKt07unFpiwsD8WIgiEe/B/gYhhc9GD6WbB6Sjiz2qzIbNjDReDQSVN98OdC/J749Mz0CoNQBZ5/P3RZ1/bQhZDv581ru5/1I1Iee7/XR0zwf0498vpegeB0/4fwmRHTolQ8HXtxP/vMC9C76vsfqpK97stefnHY9SYbsWNp1QuSdLlvxvgTrIcoRlrLoPER4jb+6XJ80v1pJtZv3e2nrPFHCz+aXd2eFvAuLInXrwhCkmB1fq78CoVc6TAQhS4LQ/zERoyp6G39mT9D5qTGnIWcEzhp/9lD4nVfoaJiSfbswQTiXI7+kfldSE0P+YiMhmDL75EcyhT9pEA8kuupDKExS2MhbOAhn7okumDpJC6XnjNW33ywms3fZ0Ugx/KZ4OfKdyafvjY0U09evS8TLExd6IEN0nN1fxOJvD+ani961jlLtUwcRxoAcybuV2nwyNJ1CPc/TnF6FutbOMq1JFWq0Fdp3PzCZQnfQJfPb29W9T9eW/lNdrR1vLgfUIOYTDihNCp/9mMFUBti7Xd8Ynm7Riax78rO9z1TNtwuRVevH3WUIVr+Q16//Aw==</diagram></mxfile>
```
Comment (karbyshevds) [inline source/img/security_comp_impl.drawio:1]: Is it required?

**Review by karbyshevds — CHANGES_REQUESTED** [review]

File: source/img/security_comp_impl.drawio (hunk 6 above)
Comment (vlad-tokarev) [reply source/img/security_comp_impl.drawio:1]: It allows you to edit the image in draw.io editor

File: source/comp_api.rst [hunk 7]
```diff
@@ -26,12 +26,10 @@ You can read all of the up-to-date documentation and invoke all methods (allowed
 Authentication and authorization
 --------------------------------
 
-:term:`API service` analyzes incoming HTTP headers for JWT token, extracts client's scopes from this token and approves / declines incoming requests based on these (provided in JWT) scopes.
+:term:`API service` distributed in :ref:`odahu-flow-core helm chart <gen_distros:Distributions>` with enabled authorization
+and pre-defined OPA policies. If :ref:`Security Subsystem <comp_security:Security-subsystem>` is installed then all requests to api service will be
```
Comment (BPylypenko) [inline source/comp_api.rst]:
```suggestion
and pre-defined OPA policies. If :ref:`Security Subsystem <comp_security:Security-subsystem>` is installed, then all requests to API service will be
```

File: source/gen_security.rst [hunk 8]
```diff
@@ -84,108 +81,41 @@ let's review the API request lifecycle and describe what is happened for each HT
...
    1. Envoy verifies JWT token in `Authorization Request Header Field`_  using `JSON Web Token (JWT) Authentication`_ filter
-   2. Envoy makes a query to OpenPolicyAgent_ as PDP_ using `External Authorization`_ filter passing parsed JWT token from the previous step
+   2. Envoy makes a query to OpenPolicyAgent_ sidecar as PDP_ using `External Authorization`_ filter passing parsed JWT token
+      from the previous step. OpenPolicyAgent_ sidecars are injected for all pods that have
+      `odahu-flow-authorization=enabled` label
 
 4. If a request is permitted by OpenPolicyAgent_, it is sent to upstream (ODAHU API)
 
 
 UML sequence diagram of successful API request described above is shown on image:
```
Comment (BPylypenko) [inline source/gen_security.rst]:
```suggestion
UML sequence diagram of a successful API request described above is shown in the image:
```

File: source/gen_security.rst [hunk 9]
```diff
@@ -84,108 +81,41 @@ let's review the API request lifecycle and describe what is happened for each HT
...
-     #. odahu_viewer
-     #. odahu_data_scientist
-     #. odahu_admin
-
+ODAHU is distributed with build-in policies that are written on `Rego policy language`_ and included into helm charts
+of appropriate services.
 
-Set user role manually
-------------------------
+`Role based access control`_ is implemented by default for next services
```
Comment (BPylypenko) [inline source/gen_security.rst]:
```suggestion
`Role-based access control`_ is implemented by default for next services
```

File: source/gen_security.rst [hunk 10]
```diff
@@ -84,108 +81,41 @@ let's review the API request lifecycle and describe what is happened for each HT
...
-In this section, different ways to manage access control in ODAHU is described
-
-
-Built-in policies overview
-===========================
-
-ODAHU is distributed with a pre-defined set of OpenPolicyAgent_ policies. These policies implement  simple
+:ref:`API <comp_api:API>` and :ref:`Feedback aggregator <comp_feedback:Feedback aggregator>` are distributed
+with a pre-defined set of OpenPolicyAgent_ policies. These policies implement  simple
 `Role based access control`_ (RBAC).
```
Comment (BPylypenko) [inline source/gen_security.rst]:
```suggestion
`Role-based access control`_ (RBAC).
```

File: source/gen_security.rst [hunk 11]
```diff
@@ -315,10 +250,11 @@ In this file, we:
 
 
 Create custom policies
-========================
+---------------------------------
 
 If `Role based access control`_ is not enough for your purposes you can customize policies to
 use more general `Access based access control`_. For this rewrite `core.rego` file or create your own rego policies
```
Comment (BPylypenko) [inline source/gen_security.rst]:
```suggestion
If `Role-based access control`_ is not enough for your purposes you can customize policies to
use more general `Attribute-based access control`_. For this purpose, rewrite `core.rego` file or create your own rego policies
```

File: source/comp_feedback.rst [hunk 12]
```diff
@@ -0,0 +1,46 @@
...
+======================
+
+:term:`Feedback aggregator` is a service that provides a :term:`Model Feedback API` and gathers input and output
+:term:`prediction requests <Model prediction API>`
+
+API-provided URLs
+--------------------------
+
+:term:`Model Feedback API` provide just single endpoint that
+allow you send feedback on a :term:`prediction request <Model prediction API>`:
```
Comment (BPylypenko) [inline source/comp_feedback.rst:13]:
```suggestion
:term:`Model Feedback API` provides just a single endpoint that
allows you send feedback on a :term:`prediction request <Model prediction API>`:
```

File: source/comp_feedback.rst [hunk 13]
```diff
@@ -0,0 +1,46 @@
...
+`POST /api/v1/feedback`
+
+Information about this URL can be viewed using the auto-generated, interactive Swagger page. It is located at ``<api-address>/swagger/index.html``.
+You can read all of the up-to-date documentation and invoke this endpoint (allowed for your account) right from this web page.
+
+Authentication and authorization
+--------------------------------
+
+:term:`Feedback aggregator` distributed in :ref:`odahu-flow-core helm chart <gen_distros:Distributions>` with enabled authorization
+and pre-defined OPA policies. If :ref:`Security Subsystem <comp_security:Security-subsystem>` is installed then all requests
```
Comment (BPylypenko) [inline source/comp_feedback.rst]:
```suggestion
and pre-defined OPA policies. If :ref:`Security Subsystem <comp_security:Security-subsystem>` is installed, then all requests
```

File: source/comp_feedback.rst [hunk 14]
```diff
@@ -0,0 +1,46 @@
...
+to :term:`Model Feedback API` service will be enforced using :ref:`pre-defined OPA policies <gen_security:ODAHU API and Feedback aggregator policies>`.
+
+Implementation details
+----------------------
+
+:term:`Feedback aggregator` contains two major subcomponents
+
+- REST Server that provides :term:`Model Feedback API` and send them to configured fluentd server
+- Envoy Proxy tap filter that catch all requests and response of deployed models and send this info
```
Comment (BPylypenko) [inline source/comp_feedback.rst]:
```suggestion
- REST Server provides :term:`Model Feedback API` and sends them to configured fluentd server
- Envoy Proxy tap filter catches all requests and responses of deployed models and sends this info
```

File: source/comp_security.rst [hunk 15]
```diff
@@ -0,0 +1,50 @@
...
+
+- Webhook server that injects `OpenPolicyAgent`_ sidecars into pods that labeled by ``odahu-flow-authorization=enabled``
+- EnvoyFilter that configures Istio-proxy sidecars located in pods labeled by ``odahu-flow-authorization=enabled``
+  to force authentication and authorization for all incoming http requests
+- ConfigMap with common policy that included into all `OpenPolicyAgent`_ sidecars and implements masking of a sensitive data
+- ConfigMap with default policy that included into `OpenPolicyAgent`_ sidecars when pod does not specify ConfigMap with its polices
+
+When webhook server deploys `OpenPolicyAgent`_ sidecar it attaches policies from ConfigMap. ConfigMap can be found by
+value of pod label ``opa-policy-config-map-name``. If this label is missed then default policy will be used.
+Default policy – reject all requests.
```
Comment (BPylypenko) [inline source/comp_security.rst]:
```suggestion
When the webhook server deploys `OpenPolicyAgent`_ sidecar, it attaches policies from ConfigMap. ConfigMap can be found by
the value of the pod label ``opa-policy-config-map-name``. If this label is missed, then the default policy will be used.
Default policy – reject all requests.
```

File: source/comp_security.rst [hunk 16]
```diff
@@ -0,0 +1,50 @@
...
+
+Different ODAHU components such as :ref:`API <comp_api:API>` and :ref:`Feedback aggregator <comp_feedback:Feedback aggregator>`
+are distributed with a pre-defined set of OpenPolicyAgent_ policies. They create ConfigMap with their policies during the deployment process.
+
+If user change ConfigMap with policies then appropriate pod must be restarted to refresh its policies.
```
Comment (BPylypenko) [inline source/comp_security.rst]:
```suggestion
If you change ConfigMap with policies then the appropriate pod must be restarted to refresh its policies.
```

File: source/gen_security.rst [hunk 17]
```diff
@@ -211,8 +141,13 @@ All policies customization can be done on the stage of system configuration as d
 :ref:`installation guide <tutorials_installation:Installation>`
 
 
+Customize
+=========================
+
+In this section, different ways to customize pre-defined policies
+
 Extend roles
```
Comment (BPylypenko) [inline source/gen_security.rst:149]:
```suggestion
Extended roles
```

File: source/gen_security.rst [hunk 18]
```diff
@@ -211,8 +141,13 @@ All policies customization can be done on the stage of system configuration as d
...
-=======================
+----------------------
 
 To define new custom roles you should just add it as a variable in file `roles.rego`
```
Comment (BPylypenko) [inline source/gen_security.rst]:
```suggestion
To define new custom roles, you should add them as a variable in the `roles.rego` file
```

**Review by BPylypenko — CHANGES_REQUESTED** [review]

File: source/gen_security.rst [hunk 19]
```diff
@@ -344,22 +280,32 @@ use more general `Access based access control`_. For this rewrite `core.rego` fi
...
 - lines 23-26: allow access to swagger docs to any user
 
+
+ODAHU ML Models pods policies
+""""""""""""""""""""""""""""""""""""""
+
+All deployed models contain default policies that permit requests to them to all users
+that have :term:`Model Deployment Access Role Name`. This role can be set on model deployment stage using
+.Spec.roleName ModelDeployment manifest key and also some other roles that statically configured in polices
+when ODAHU is deployed
```
Comment (BPylypenko) [inline source/gen_security.rst]:
```suggestion
All deployed models contain default policies that permit requests to them for all users
that have :term:`Model Deployment Access Role Name`. This role can be set at the model deployment stage using
.Spec.roleName key of the ModelDeployment manifest and also statically configured in policies
during ODAHU deployment.
```

**Review by keshamin — APPROVED** [review]

**Review by BPylypenko — APPROVED** [review]

**Pull request merged**